lib/composables/Url.ts

```diff
@@ -39,11 +44,14 @@ export function useUrlQuerySync(
       flattenState[key] = cast ? cast(value) : value
     })
 
-    deepAssign(state, unflattenObject(flattenState))
+    deepAssign(
+      isRef<Record<string, any>>(state) ? state.value : state,
```

Comment:
Just curious, probably I'm missing something but can't we use `toValue` here?

Reply:
Yeah I was thinking that too. I'm not sure if assignments work with it. I'll take a look.

Reply:
I initially thought assignment might not work with that. Seems to be working fine. Updated it.

Reply:
Nice. Yeah I was curious too so went ahead and checked the code but this ternary is what exactly Vue is doing 👍 
https://github.com/vuejs/core/blob/main/packages/reactivity/src/ref.ts

```ts
export function toValue<T>(source: MaybeRefOrGetter<T> | ComputedRef<T>): T {
  return isFunction(source) ? source() : unref(source)
}

export function unref<T>(ref: MaybeRef<T> | ComputedRef<T>): T {
  return isRef(ref) ? ref.value : ref
}
```

Reply:
Yeah. I was just looking at that too 😅 We don't need toValue either. unref should work fine here and avoid extra checks for function.

Reply:
Very good point.